src/components/EditableToDo.vue

```diff
@@ -0,0 +1,195 @@
+                  {{ priority }}
+                </p>
+              </li>
+            </ul>
+          </div>
+        </div>
+      </div>
+      <div class="flex flex-row space-x-1" :class="currentOpacity">
+        <img src="../assets/icons/Vectorcalendar.svg" />
+        <p class="text-xs">{{ formatShortDate(date) }}</p>
+      </div>
+      <div class="flex mt-4 mb-4">
+        <textarea
+          v-model="currentText"
+          placeholder="Lorem ipsum"
+          class="border-none h-18 w-56 lg:h-28 lg:w-96 text-start text-sm lg:text-lg text-grey"
+          :class="currentOpacity"
+          :disabled="isDropDownShown"
+        ></textarea>
+      </div>
+      <div class="text-sm flex flex-row font-semibold space-x-2">
+        <PrimaryButton
+          class="bg-green"
+          :class="currentOpacity"
+          :disabled="isDropDownShown"
+          @click="saveToDo"
+          >Save</PrimaryButton
+        >
+        <PrimaryButton
+          class="bg-buttonGray text-black"
+          :class="currentOpacity"
+          :disabled="isDropDownShown"
+          @click="deleteToDo"
+          >Delete</PrimaryButton
+        >
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+  import { computed, ref, PropType } from 'vue';
+  import { Priority, ToDoItem } from '../models/todoitem-model';
+  import { formatShortDate } from '../utils/date-formatting';
+  import { getItemPriority } from '../utils/item-priority';
+  import ConfirmationModal from './ConfirmationModal.vue';
+  import PrimaryButton from './PrimaryButton.vue';
+
+  const props = defineProps({
+    item: Object as PropType<ToDoItem>,
+    index: {
+      type: Number,
+      default: -1,
+    },
+  });
+
+  const emit = defineEmits<{
+    (e: 'removeItem', ind: Number): void;
+    (e: 'addToDo', newToDo: ToDoItem): void;
+    (e: 'hideForm'): void;
+    (e: 'updateToDo', ind: Number, newToDo: ToDoItem): void;
+  }>();
+
+  const isDropDownShown = ref(false);
+
+  const currentOpacity = computed(() => { return isDropDownShown.value ? 'opacity-30' as const : '' as const;});
+
+  const currentTitle = ref(props.item?.title || '');
+
+  const currentText = ref(props.item?.text || '');
+
+  const currentPriority = ref(props.item?.priority || Priority.low);
+
+  const date = new Date();
+
+  const priorityList = [Priority.low, Priority.medium, Priority.high];
+
+  const isPopUpShown = ref(false);
+
+  const textColorOfDropDown = computed(() => { return isDropDownShown.value ? 'text-black' as const : '' as const;});
+
+  const positionOfDropDownButton = computed(() =>
+    isPopUpShown.value ? '' : 'relative'
+  );
+
+  const dropDownStyle = computed(() => {
+    if (isDropDownShown.value === true) {
+      return 'text-black bg-white border-2 border-black' as const;
+    }
+    return getItemPriority(currentPriority.value);
+  });
+
+  function styleOfPriorityButton(ind: number) {
+    if (currentPriority.value === priorityList[ind]) {
```

Comment:
You can use ternary instead of the `if` statement, the code will be shorter:
`return currentPriority.value === priorityList[ind] ? 'border border-black' as const : ''`